datalad/utils.py

```diff
@@ -921,6 +1002,8 @@ def unique(seq, key=None, reverse=False):
     return out[::-1] if reverse else out
 
 
+# TODO: Annotate (would be made easier if the return value was always a dict
+# instead of doing `v.__class__(...)`)
 def map_items(func, v):
```

Comment:
well, true, but IMHO it would be nice to keep it generic. May be there is some `Mappable` type "behind" `dict` which could be used here so anything with `.items()` kinda would be usable with it?

if not -- we could restrict to `dict` in type annotation (while adding a comment that in principle could be applied to any mappable thing with `.items()`) -- there is only 1 use of this helper and it is on `dict`.

Reply:
Note that the tests for this function specifically check that it returns the input type when it's a custom class.